src/opm/input/eclipse/EclipseState/Grid/FieldProps.cpp

```diff
@@ -905,7 +906,11 @@ void FieldProps::handle_region_operation(const DeckKeyword& keyword) {
                   with global storage.
                 */
                 if (field_data.global_data)
-                    throw std::logic_error("Region operations on 3D fields with global storage is not implemented");
+                {
+                    const auto& location = keyword.location();
+                    throw std::logic_error(fmt::format("In file {} line {}: {} region operation on 3D field {} with global storage is not implemented!",
+                                                       location.filename, std::to_string(location.lineno), keyword.name(), target_kw));
```

Comment:
This is not an input rrror, but rather insufficient parser capabilities. Hence Why should we classify it as one?

Reply:
>> would it make sense to use `OpmInputError` directly here instead of `std::logic_error`?
>
> This is not an input error, but rather insufficient parser capabilities

Then it's arguably not a `logic_error` either.  It is, at most, a `runtime_error` instead.

> Why should we classify [this condition] as [an `OpmInputError`]?

To get more uniform diagnostic messages and better treatment of the `Location`.

Reply:
> To get more uniform diagnostic messages and better treatment of the Location.

Sorry, but I don't see this in the code. Could be my poor skills.

Seems like the only reason is to indicate that we think the error message is good (and cannot be improved) and hence the error will not be caught and (somewhen?) rethrown as an OpmInputError.
But as this seems to be the case here, I'll try if this helps (always annoyed by seeing errors multiple times).

Reply:
Seems like the unification is done when catching. Documentation slightly missleading.